[13.0[[MIG] hr_timesheet_task_stage: Migration to 13.0

manuelcalerosolis opened the pull request:
This is the migration to v 13.0 module hr_timesheet_task_stage

@Tecnativa TT19750

alexey-pelykh committed:
[ADD] hr_timesheet_task_stage: control task stage from timesheet

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (8 of 8 strings)

Translation: timesheet-12.0/timesheet-12.0-hr_timesheet_task_stage
Translate-URL: https://translation.odoo-community.org/projects/timesheet-12-0/timesheet-12-0-hr_timesheet_task_stage/de/

yajo committed:
[FIX] hr_timesheet_task_stage: Disable automatic hotkeys

Without this attribute, Odoo will try to assign a hotkey to buttons. It's very unlikely that these buttons have to be used like this, and since there's a limited number of available hotkeys, they steal them to more important ones like the wizard proposed at https://github.com/OCA/project/pull/586.

yajo commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Don't commit this file.

yajo commented on hr_timesheet_task_stage/oca_dependencies.txt:
```diff
@@ -0,0 +1 @@
+project https://github.com/OCA/project project_stage_closed
```
You only need this:

```suggestion
project
```

manuelcalerosolis replied:
ok

yajo replied:
But... you have to upload the change.

Also, this file must be in the repo root, not in the addon folder.

pedrobaeza commented:
I don't see the commit history nor the black/isort thing.

Commit:
[IMP] hr_timesheet_task_stage: black, isort

Commit:
[MIG] hr_timesheet_task_stage: Migration to 13.0

	modificado:     hr_timesheet_task_stage/models/account_analytic_line.py
	modificado:     hr_timesheet_task_stage/tests/test_hr_timesheet_task_stage.py

pedrobaeza commented on hr_timesheet_task_stage/__manifest__.py:
```diff
@@ -7,7 +7,7 @@
 {
     "name": "Task Log: Open/Close Task",
-    "version": "12.0.1.1.0",
+    "version": "13.0.1.1.0",
```
```suggestion
    "version": "13.0.1.0.0",
```

yajo replied:
This is not resolved yet.

ernestotejeda commented on hr_timesheet_task_stage/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "version": "13.0.1.1.0",
+    "category": "Operations/Timesheets",
+    "website": "https://github.com/OCA/timesheet",
+    "author": "Tecnativa, " "Brainbean Apps, " "Odoo Community Association (OCA)",
```
```suggestion
    "author": "Tecnativa, Brainbean Apps, Odoo Community Association (OCA)",
```

ernestotejeda commented on hr_timesheet_task_stage/models/account_analytic_line.py:
```diff
@@ -0,0 +1,63 @@
+        ProjectTaskType = self.env["project.task.type"]
+
+        for line in self.filtered("task_id.project_id"):
+            if line.task_id.project_id:
```
```suggestion
```
I think this condition is not necessary because lines have already been filtered in the previous step

ernestotejeda reviewed (commented):
Thanks @manuelcalerosolis

ernestotejeda commented:
@pedrobaeza , Is there any runbot where I can test this functionally?

pedrobaeza commented:
See in their section in runbot: https://runbot.odoo-community.org/runbot/repo/https-github-com-oca-timesheet-117

ernestotejeda commented:
I had searched there already but no runbot appears either

ernestotejeda commented:
I'm going to test it locally

pedrobaeza commented:
This is probably because there are not enough free slots. It's a problem that @gurneyalex is looking to.

ernestotejeda reviewed (changes requested):
The button to open/close a task from account.analytic.line form view doesn't look good

![Peek 2020-02-11 11-37](https://user-images.githubusercontent.com/38267832/74257622-1a316580-4cc3-11ea-97e3-34c54d4a317b.gif)

Saran440 commented:
@manuelcalerosolis Could you please resolve Conflicting files and Code reviewed?

AntoniRomera commented:
Hello @manuelcalerosolis,

Do you want to continue the migration of the module? I see that it would be necessary to fix the conflicts. Can I help with something?

manuelcalerosolis commented:
> Hello @manuelcalerosolis,
> 
> Do you want to continue the migration of the module? I see that it would be necessary to fix the conflicts. Can I help with something?

@AntoniRomera I no longer work at Tecnativa, so I have no possibility to change this PR.

pedrobaeza commented:
I will fix it and merge.

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump
